daemon/cmd/policy.go

```diff
@@ -355,19 +336,16 @@ func (d *Daemon) policyAdd(sourceRules policyAPI.Rules, opts *policy.AddOptions,
 	// processing in this function.
 	source := ""
 	if opts != nil {
-		source = opts.Source
+		source = string(opts.Source)
 	}
 	d.endpointManager.CallbackForEndpointsAtPolicyRev(d.ctx, newRev, func(now time.Time) {
 		duration, _ := safetime.TimeSinceSafe(policyAddStartTime, logger)
 		metrics.PolicyImplementationDelay.WithLabelValues(source).Observe(duration.Seconds())
 	})
 
-	// remove prefixes of replaced rules above. Refcounts have been incremented
-	// above, so any decrements here will be no-ops for CIDRs that are re-added,
-	// and will trigger deletions for those that are no longer used.
+	// Remove prefixes of replaced rules above from the d.prefixLengths tracker.
 	if len(removedPrefixes) > 0 {
 		logger.WithField("prefixes", removedPrefixes).Debug("Decrementing replaced CIDR refcounts when adding rules")
-		d.ipcache.ReleaseCIDRIdentitiesByCIDR(removedPrefixes)
 		d.prefixLengths.Delete(removedPrefixes)
```

Comment:
Why is it safe to ignore the return value from `Delete()` here?

Reply:
I think this is probably just a long ago corner cut where basically we used to track unique prefix lengths in CIDR policies (eg `/32`, `/31`,, ...) so that if we ever added a new prefix length, we could generate hashmap lookup code in the datapath to fake LPM properties. I suspect we never implemented cleanup logic for these. The consequence would be that if you added a /32, /31, /30 CIDR policy then remove the /32, we'd still do lookups for /32 even though you don't have any policies using that prefix length. Still correct, just slightly slower.

That code however is no longer used, it was only for Linux 4.9 since that didn't have BPF LPM maps, so this code is dead at this point. https://github.com/cilium/cilium/issues/22116 has a tracking item to remove it.